Core/include/Acts/Seeding/CandidatesForSpM.hpp

```diff
@@ -9,141 +9,136 @@
-    
-    void addToCollection(std::vector< value_type >&,
-			 sp_type& SpB, sp_type& SpT, float weight, float zOrigin,
-			 bool isQuality);
-    void insertToCollection(std::vector< value_type >&,
-			    sp_type& SpB, sp_type& SpT, float weight, float zOrigin,
-			    bool isQuality);
-
-  public:
-    // sizes
-    std::size_t m_max_size_high;
-    std::size_t m_max_size_low;
-    std::size_t m_n_high;
-    std::size_t m_n_low;
-
-    // space points
-    sp_type m_SpB;
-    sp_type m_SpM;
-
-    // storage
-    // These vectors are sorted as a min heap tree
-    // Each node is lower then its childs
-    // Thus, it is guaranteed that the lower elements is at the front
-    // Sorting criteria is the seed quality 
-
-    // storage for candidates with high quality
-    std::vector< value_type > m_storage_high;
-    // storage for candidates with low quality
-    std::vector< value_type > m_storage_low;
-  };
-
-  template<typename external_space_point_t>
-  inline
-  const typename CandidatesForSpM<external_space_point_t>::sp_type&
-  CandidatesForSpM<external_space_point_t>::spM() const
-  { return m_SpM; }
-  
-  template<typename external_space_point_t>
-  inline void CandidatesForSpM<external_space_point_t>::setMaxElements(std::size_t n_low,
-								       std::size_t n_high)
-  {
-    if (m_storage_high.capacity() < n_high) m_storage_high.reserve(n_high);
-    if (m_storage_low.capacity() < n_low) m_storage_low.reserve(n_low);
-    m_max_size_high = n_high;
-    m_max_size_low = n_low;
-  }
-
-  template<typename external_space_point_t>
-  inline void CandidatesForSpM<external_space_point_t>::setMediumSp(typename CandidatesForSpM<external_space_point_t>::sp_type idx)
-  { m_SpM = idx; }
-
-  template<typename external_space_point_t>
-  inline void CandidatesForSpM<external_space_point_t>::setBottomSp(typename CandidatesForSpM<external_space_point_t>::sp_type idx)
-  { m_SpB = idx; }
-
-  template<typename external_space_point_t>
-  inline float CandidatesForSpM<external_space_point_t>::top(const std::vector<value_type>& storage) const
-  { return weight(storage, 0); }
-
-  template<typename external_space_point_t>
-  inline bool CandidatesForSpM<external_space_point_t>::exists(std::size_t n, std::size_t max_size) const
-  { return n < max_size; }
-
-  template<typename external_space_point_t>
-  inline float CandidatesForSpM<external_space_point_t>::weight(const std::vector<value_type>& storage, std::size_t n) const
-  { return std::get<InternalComponents::I_WEIGHT>(storage[n]); }
-
-  template<typename external_space_point_t>
-  inline void CandidatesForSpM<external_space_point_t>::clear()
-  {
-    // do not clear max size, this is set only once
-    m_n_high = 0;
-    m_n_low = 0;
-    // clean fixed space points
-    m_SpB = default_value;
-    m_SpM = default_value;
-    // clean storage
-    m_storage_high.clear();
-    m_storage_low.clear();
-  }
-  
+template <typename external_space_point_t>
```

Comment:
This class would benefit from more documentation.

Reply:
will add